src/main/java/budgetbuddy/model/AccountsManager.java

```diff
@@ -5,26 +5,31 @@
 import java.util.List;
 import java.util.function.Predicate;
 
+import budgetbuddy.commons.core.index.Index;
 import budgetbuddy.model.account.Account;
 import budgetbuddy.model.account.UniqueAccountList;
 import budgetbuddy.model.account.exception.AccountNotFoundException;
-import budgetbuddy.model.attributes.Name;
 import budgetbuddy.model.transaction.Transaction;
+import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 import javafx.collections.transformation.FilteredList;
 
 /**
- * Manages the loans of each person in a list of persons.
+ * Manages the account in a list of accounts.
  */
 public class AccountsManager {
 
     private final UniqueAccountList accounts;
 
     private final FilteredList<Account> filteredAccounts;
 
+    private final ObservableList<Account> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Account> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
```

Comment:
Not sure why you have both a UniqueAccountList as well as an internalList+internalUnmodifiableList